engines/instance_verification/lib/instance_verification/engine.rb

```diff
@@ -1,9 +1,32 @@
+require 'fileutils'
+
 module InstanceVerification
-  def self.update_cache(remote_ip, system_login, product_id, is_byos)
+  def self.update_cache(remote_ip, system_login, product_id, _is_byos)
+    # TODO: BYOS scenario
+    # to be addressed on a different PR
+    cache_config_data = InstanceVerification.cache_config
     cache_key = [remote_ip, system_login, product_id].join('-')
     # caches verification result to be used by zypper auth plugin
-    expire_cache_time = is_byos ? 24.hours : 20.minutes
-    Rails.cache.write(cache_key, true, expires_in: expire_cache_time)
+    InstanceVerification.write_cache_file(
+      cache_config_data['REPOSITORY_CLIENT_CACHE_DIRECTORY'],
+      cache_key
+    )
+  end
+
+  def self.cache_config
+    cache_config_data = {}
+    File.open(Rails.application.config.cache_config_file, 'r') do |cache_config_file|
+      cache_config_file.each_line do |line|
+        line_data = line.split(/=|\n/)
+        cache_config_data[line_data[0]] = line_data[1]
+      end
+    end
+    cache_config_data
+  end
+
+  def self.write_cache_file(cache_dir, cache_key)
+    FileUtils.mkdir_p(cache_dir)
```

Comment:
We shouldn't try to make the directory every time we want to write to the cache.